src/services/Models.service.tsx

```diff
@@ -222,38 +269,57 @@ const dracoLoader = new DRACOLoader();
 dracoLoader.setDecoderPath('https://www.gstatic.com/draco/versioned/decoders/1.5.5/');
 loader.setDRACOLoader(dracoLoader);
 
+async function loadGltf(source: Blob | string): Promise<GLTF> {
+  let gltf: GLTF;
+
+  if (source instanceof Blob) {
+    const url = URL.createObjectURL(source);
+    try {
+      gltf = await loader.loadAsync(url);
+    } finally {
+      URL.revokeObjectURL(url);
+    }
+  } else {
+    gltf = await loader.loadAsync(source);
+  }
+
+  return gltf;
+}
+
 export const useGltfLoader = (source: Blob | string): GLTF =>
-  suspend(
-    async () => {
-      if (source instanceof Blob) {
-        const buffer = await source.arrayBuffer();
-        return (await loader.parseAsync(buffer, '')) as unknown as GLTF;
+  suspend(async () => loadGltf(source), [source], { lifespan: 100 });
+
+export const useGltfCachedLoader = (source: Blob | string): GLTF => {
+  const cachedGltf = useRef<GLTF | null>(null);
+  const prevSource = useRef<Blob | string | null>(null);
+
+  const scene = suspend(
+    async (): Promise<GLTF> => {
+      if (source === prevSource.current && cachedGltf.current) {
+        return cachedGltf.current;
       }
 
-      return loader.loadAsync(source);
+      const gltf = await loadGltf(source);
+
+      cachedGltf.current = gltf;
+      prevSource.current = source;
+
+      return gltf;
     },
     [source],
     { lifespan: 100 }
   );
 
-export const useGltfCachedLoader = (source: Blob | string): GLTF => {
-  const cachedGltf = useRef<Map<string, GLTF>>(new Map<string, GLTF>());
-
-  return suspend(async (): Promise<GLTF> => {
-    if (cachedGltf.current.has(source as string)) {
-      return cachedGltf.current.get(source as string)!;
-    }
-    let result: GLTF;
-    if (source instanceof Blob) {
-      const buffer = await source.arrayBuffer();
-      result = (await loader.parseAsync(buffer, '')) as GLTF;
-    } else {
-      result = await loader.loadAsync(source);
-    }
+  useEffect(
```

Comment:
@ericjansenrpm , tested putting the dispose at line 303 but there were cases were the fallback would be disposed before the next fully loaded. So keeping this here for now!